libp2p/NodeTable.h

```diff
@@ -150,7 +150,9 @@ class NodeTable : UDPSocketEvents
     /// proof for it or it has been restored from the network config), also add it to the node table.
     /// If the node is unknown (i.e. we haven't completed the endpoint proof for it yet) then ping
     /// it to trigger the endpoint proof.
-    void addNode(Node const& _node, NodeRelation _relation = NodeRelation::Unknown);
+    ///
+    /// @return Pointer to added entry on null if node not added.
+    NodeEntry* addNode(Node const& _node, NodeRelation _relation = NodeRelation::Unknown);
```

Comment:
I'd prefer this not to be a raw pointer, especially because it's a public method of the class. Either return bool and lookup once again (don't bother micro-optimizing) or return shared_ptr

Reply:
Changed to `bool`.